llvm/lib/CAS/OnDiskCAS.cpp

```diff
@@ -98,7 +98,11 @@ std::optional<ObjectRef> OnDiskCAS::getReference(const CASID &ID) const {
 }
 
 Expected<bool> OnDiskCAS::isMaterialized(ObjectRef ExternalRef) const {
-  return DB->containsObject(convertRef(ExternalRef));
+  Expected<std::optional<ondisk::ObjectHandle>> ObjHnd =
+      DB->load(convertRef(ExternalRef));
```

Comment:
I'd suggest to avoid `load` to make `isMaterialized` a bit more efficient (`load` reads standalone files into memory buffers).

How about introducing `OnDiskGraphDB::isMaterialized` which would check for `contains` in both chained DBs and "fault-in" when necessary? You could add an internal variant of `OnDiskGraphDB::containsObject` that not only returns true/false but which DB contained it, which would get called by the existing `OnDiskGraphDB::containsObject` function along with the new `OnDiskGraphDB::isMaterialized`.